src/UIKit/UIApplication.cs

```diff
@@ -57,33 +60,31 @@ internal static void Initialize ()
 		}
 		
 #if !WATCH
-		public static void Main (string [] args, string principalClassName, string delegateClassName)
+		[Obsolete ("Use the overload with 'Type' instead of 'String' parameters for type safety.")]
+		public static void Main (string []? args, string? principalClassName, string? delegateClassName)
 		{
-			var p = NSString.CreateNative (principalClassName);
-			var d = NSString.CreateNative (delegateClassName);
-			try {
-				Main (args, p, d);
-			} finally {
-				// it just looks nicer to release them
-				NSString.ReleaseNative (d);
-				NSString.ReleaseNative (p);
-			}
+			var p = CFString.CreateNative (principalClassName);
+			var d = CFString.CreateNative (delegateClassName);
+			Initialize ();
+			UIApplicationMain (args == null ? 0 : args.Length, args, p, d);
+			CFString.ReleaseNative (d);
+			CFString.ReleaseNative (p);
 		}
 		
-		public static void Main (string [] args, Type principalClass, Type delegateClass)
+		public static void Main (string []? args, Type? principalClass, Type? delegateClass)
 		{
-			Main (args, principalClass == null ? null : new Class (principalClass).Name, delegateClass == null ? null : new Class (delegateClass).Name);
-		}
-
-		public static void Main (string [] args)
-		{
-			Main (args, IntPtr.Zero, IntPtr.Zero);
+			var p = principalClass == null ? IntPtr.Zero : CFString.CreateNative (new Class (principalClass).Name);
+			var d = delegateClass == null ? IntPtr.Zero : CFString.CreateNative (new Class (delegateClass).Name);
+			Initialize ();
+			UIApplicationMain (args == null ? 0 : args.Length, args, p, d);
+			CFString.ReleaseNative (d);
+			CFString.ReleaseNative (p);
 		}
 
-		static void Main (string [] args, IntPtr principal, IntPtr @delegate)
+		public static void Main (string []? args)
 		{
 			Initialize ();
-			UIApplicationMain (args.Length, args, principal, @delegate);
+			UIApplicationMain (args == null ? 0 : args.Length, args, IntPtr.Zero, IntPtr.Zero);
```

Comment:
Same, a little simpler to parse:
```suggestion
			UIApplicationMain (args?.Length  ?? 0, args, IntPtr.Zero, IntPtr.Zero);
```